### PR TITLE
Add Analytical Entropy

### PR DESCRIPTION
**Pull request by TMats**
We may want an analytical version of entropy and cross-entropy.

Question: Is it better to use `SetLoss` instead of `Loss`? (I don't know it is suitable for this case.)

### CODE REVIEW

**Comment by TMats** [comment]
something is wrong with batch
```
In [33]: loc = torch.tensor(0.)
    ...: scale = torch.tensor(1.)
    ...: prior = Normal(loc=loc, scale=scale, var=["z"], dim=64, name="p_prior")

In [34]: Entropy(prior).eval()
Out[34]: tensor([87.5932])

In [35]: AnalyticalEntropy(prior).eval()
Out[35]: tensor(1.4189)
```

**Comment by TMats** [comment]
If prior has the batch axis this problem won't happen (not related to this implementation)
```
>>> loc = torch.zeros([1,64])
>>> scale = torch.ones([1,64])
>>> prior = Normal(loc=loc, scale=scale, var=["z"], dim=64, name="p_prior")
>>> AnalyticalEntropy(prior).eval()
tensor([90.8121])
```

**Comment by masa-su** [comment]
Thank you for your pull request!

> Question: Is it better to use SetLoss instead of Loss? (I don't know it is suitable for this case.)
I think you don't need to inherit `SetLoss` in this case because the `entropy` method is not implemented in Distribution API for now.

**Comment by masa-su** [comment]
@TMats 
Also, I would be glad if you could add your implemented class to the following docs.
https://github.com/masa-su/pixyz/blob/develop/v0.1.0/docs/source/losses.rst
